db/migrate/20220613211542_add_current_sign_in_at.rb

```diff
@@ -0,0 +1,11 @@
+class AddCurrentSignInAt < ActiveRecord::Migration[7.0]
+  def change
+    change_table :users do |t| 
+      t.integer :sign_in_count, :default => 0 
+      t.datetime  :current_sign_in_at
+      t.datetime :last_sign_in_at 
+      t.string :current_sign_in_ip 
+      t.string :last_sign_in_ip
+    end
+  end
+end
```

Comment:
![image](https://user-images.githubusercontent.com/65068771/173459023-09f1836e-1db7-439f-9235-a07c753d8c3d.png)

- [ ] Kindly fix this error. The way I have analyzed your code and the approach you used I have realised that some of your migrations are not needed so far because all of them are already inside the main devise migration(`db/migrate/20220613193104_add_devise_to_users.rb`) you just need to uncomment their logics.

these are those migrations not needed. kindly delete them.

![image](https://user-images.githubusercontent.com/65068771/173459626-aad2c925-2b95-461d-86d4-3ea979acae2b.png)

Hint: now if you want to add `confirmable`, `Trackable`, `Lockable` ... to devise kindly uncomment theirs logic in the  the main devise migration(`db/migrate/20220613193104_add_devise_to_users.rb`). Here is how your only migration for devise will be :point_down: 

```rb 
# frozen_string_literal: true

class AddDeviseToUsers < ActiveRecord::Migration[7.0]
  def self.up
    change_table :users do |t|
      ## Database authenticatable
      t.string :email,              null: false, default: ""
      t.string :encrypted_password, null: false, default: ""

      ## Recoverable
      t.string   :reset_password_token
      t.datetime :reset_password_sent_at

      ## Rememberable
      t.datetime :remember_created_at

      # Trackable
      t.integer  :sign_in_count, default: 0, null: false
      t.datetime :current_sign_in_at
      t.datetime :last_sign_in_at
      t.string   :current_sign_in_ip
      t.string   :last_sign_in_ip

      # Confirmable
      t.string   :confirmation_token
      t.datetime :confirmed_at
      t.datetime :confirmation_sent_at
      t.string   :unconfirmed_email # Only if using reconfirmable

      # Lockable
      t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
      t.string   :unlock_token # Only if unlock strategy is :email or :both
      t.datetime :locked_at


      # Uncomment below if timestamps were not included in your original model.
      # t.timestamps null: false
    end

    add_index :users, :email,                unique: true
    add_index :users, :reset_password_token, unique: true
    # add_index :users, :confirmation_token,   unique: true
    # add_index :users, :unlock_token,         unique: true
  end

  def self.down
    # By default, we don't want to make any assumption about how to roll back a migration when your
    # model already existed. Please edit below which fields you would like to remove in this migration.
    raise ActiveRecord::IrreversibleMigration
  end
end
```